### DISCUSSION TITLE
`airflow tasks test` break with execution date w/o time and offset

### DISCUSSION THREAD

**Discussion by pogson** [apache/airflow#17297] [Q&A]
TL;DR  Why does `airflow tasks test DAG TASK '2021-05-01'` break but `airflow tasks test DAG TASK '2021-05-01T05:00:00+00:00'` works fine?

For the last while I was struggling to find out why my tasks works when scheduled on the worker but not when running `airflow tasks test`.

It was failing with:
```sqlalchemy.exc.DataError: (MySQLdb._exceptions.DataError) (1406, "Data too long for column 'value' at row 1")```

After digging in I found the xcom value was `1120` characters after picking when scheduled but when run from `airflow test` a whopping `96295` characters.

I eventually found that things would work if I supplied an execution date with a time and offset. The value I'm passing is a simple dictionary that contains a couple date times that are derived by manipulating the execution date of the DAG. It looks like the datetime value is pickled differently somehow and maybe more of a pendulum/pickle question.

This isn't a big deal since I can just supply a proper execution date which I probably should be doing anyway but I was being lazy.

Curious if anyone else has seen this or understands why it's happening?

**Comment by pogson** [comment]
Here is a sample dag that could be used to reproduce the problem.

```from airflow import DAG
from airflow.utils import timezone
from airflow.operators.python_operator import PythonOperator

default_args = {
    'owner':'me',
    'start_date': timezone.datetime(2019,1,1), # This is the earliest date this dag can run.
    'provide_context': True, # We want our tasks to receive context by default
}

dag = DAG(
    'broken', # The name of our DAG
    default_args=default_args,
    schedule_interval=None
)

with dag:
    fail = PythonOperator(
        task_id='fail',
        python_callable=lambda **context: {"date": context['execution_date']}
    )
```
